### ISSUE TITLE
Add-on for openjdk imagestream and template

### ISSUE THREAD

**Issue by jorgemoralespou** [minishift/minishift#780]
As we have an EAP7 addon, we need to have one for openjdk image which is the image we promote to develop Java microservices on openshift.

**Comment by budhrg** [comment]
@jorgemoralespou  any idea where we can get the template for same ?

**Comment by jorgemoralespou** [comment]
@budhrg yes. Same place you should get for EAP, as it's probably the most trustable source:

https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_examples/files/examples/v1.5

Imagestream: https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/v1.5/xpaas-streams/jboss-image-streams.json#L371-L395

Template: https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/v1.5/xpaas-templates/openjdk18-web-basic-s2i.json

**Comment by praveenkumar** [comment]
@hferentschik @jorgemoralespou Since imagestream is same for EAP and openjdk `jboss-image-streams.json` should we add openjdk template as part of EAP addon or create new then duplicate image stream. I don't want to take a slice of imagestream ourself because then it's hard to maintain.

**Comment by jorgemoralespou** [comment]
@praveenkumar If in the EAP add-on your installing more than EAP maybe the name should be changed. I use xPaaS, or jboss or jboss-mdw. Anything that reflects what will be installed.

**Comment by praveenkumar** [comment]
@jorgemoralespou Ack so I will rename EAP addon to xPasS and add required template I think that is good way to go.

**Comment by hferentschik** [comment]
@jorgemoralespou @praveenkumar - are we sure openjdk18-web-basic-s2i.json is ready for prime time. I tried to use the imported template, but I am not able to get a running app. 

    --> Scaling openjdk-app-1 to 1
    --> Waiting up to 10m0s for pods in rc openjdk-app-1 to become ready
    error: update acceptor rejected openjdk-app-1: pods for rc "openjdk-app-1" took longer than 600 seconds to become ready

Also the meta information seems odd:

![web](https://cloud.githubusercontent.com/assets/296354/25946494/b0b7348a-364b-11e7-95a6-43b790a5654f.png)

Why is there s2i in the name and no description? 

Form the other Java based templates  I can spin up a demo app, but not from this one, even though afaict it should use https://github.com/jboss-openshift/openshift-quickstarts

**Comment by jorgemoralespou** [comment]
That's a template. You should also have an imagestreams (builder) that
provides what you mention.
What add-on are you trying? Is there a link I can use to give you a better
answer?

**Comment by hferentschik** [comment]
@jorgemoralespou, I just tested this pull request - https://github.com/minishift/minishift/pull/875 - which relates to this issue.

**Comment by jorgemoralespou** [comment]
That's missing the imagestream. It's only the template.
The imagestream used in the PR is rather old.

The latest one can be found here
<https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/v1.5/xpaas-streams/jboss-image-streams.json>
(also organized per release). As you can see there is an openjdk at the
very bottom.

I would consider this repository as the place to get the
imagestreams/templates as it's the most curated place.

2017-05-11 16:44 GMT+02:00 Hardy Ferentschik <notifications@github.com>:

**Comment by hferentschik** [comment]
> That's missing the imagestream. It's only the template.

Ahh, cool. I thought there must have been something wrong. @praveenkumar, can you have another look at this?

**Comment by praveenkumar** [comment]
@hferentschik Updated imagestream and also url description in addon. Able to deploy basic openjdk-webapp and worked as expected.

**Comment by gbraad** [comment]
Does this allow deploying CoolStore? 

```
oc process -f https://raw.githubusercontent.com/jbossdemocentral/coolstore-microservice/1.0.x/openshift/coolstore-template.yaml | oc create -f -
```

**Comment by hferentschik** [comment]
Merged via pull request #875
